src/validation.h

```diff
@@ -181,12 +181,41 @@ void UnlinkPrunedFiles(const std::set<int>& setFilesToPrune);
 /** Prune block files up to a given height */
 void PruneBlockFilesManual(int nManualPruneHeight);
 
-/** (try to) add transaction to memory pool
- * plTxnReplaced will be appended to with all transactions replaced from mempool
- * @param[out] fee_out optional argument to return tx fee to the caller **/
-bool AcceptToMemoryPool(CTxMemPool& pool, TxValidationState &state, const CTransactionRef &tx,
-                        std::list<CTransactionRef>* plTxnReplaced,
-                        bool bypass_limits, bool test_accept=false, CAmount* fee_out=nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+/**
+* Validation result for a single transaction mempool acceptance.
+* When m_accepted = true, m_replaced_transactions contains a list
+* of replaced transactions and m_base_fees contains the tx fees.
+* When m_accepted = nullopt, the transaction was not fully validated
+* (may be valid or invalid), e.g. when it is part of a package that
+* failed validation for an earlier transaction.
+*/
+struct MempoolAcceptResult {
+    // Tri-state
+    std::optional<bool> m_accepted;
+    TxValidationState m_state;
+
+    // Valid when m_accepted = true
+    std::optional<std::list<CTransactionRef>> m_replaced_transactions;
+    std::optional<CAmount> m_base_fees;
+
+    /** Constructor for failure or unfinished case */
+    explicit MempoolAcceptResult(TxValidationState state, bool finished=true) :
+        m_state(state), m_replaced_transactions(nullopt), m_base_fees(nullopt) {
+            m_accepted = finished ? std::optional<bool>(false) : nullopt;
+        }
+
+    /** Constructor for success case */
+    explicit MempoolAcceptResult(TxValidationState state,
+                                 std::list<CTransactionRef>&& replaced_txns, CAmount fees) :
+        m_accepted(true), m_state(state),
+        m_replaced_transactions(std::move(replaced_txns)), m_base_fees(fees) {}
+};
+
+/**
+ * (Try to) add a transaction to the memory pool.
```

Comment:
abc6ff1

"Evaluate a transaction mempool acceptance. If `test_accept`= false, add the transaction to the mempool".

Reply:
Added!

Reply:
To be precise, I think `bypass_limits` also overrides some feerate check (L729 in `src/validation.cpp`).

Reply:
Addressed